src/foam/nanos/auth/CreatedAwareDAO.js

```diff
@@ -33,6 +33,7 @@ foam.CLASS({
       javaCode: `
         // only set created if object does not exist in DAO yet
         if ( obj instanceof CreatedAware && getDelegate().find_(x, obj) == null ) {
+          if ( obj.isFrozen() ) obj = obj.fclone();
```

Comment:
If we do this here, then all DAO's that modify the incoming object will need to do the same check. Better to not do it and just require the uses clone() before put()'ing themselves.